module/reports/helper.js

```diff
@@ -7,736 +7,884 @@
+				// 	projectReport[CONSTANTS.common.NOT_STARTED_STATUS] = projectReport[CONSTANTS.common.STARTED]
+				// 	delete projectReport[CONSTANTS.common.SUBMITTED_STATUS]
+				// 	delete projectReport[CONSTANTS.common.STARTED]
+				// }
+
+				//Initially not supporting pdf report, hence resolving response here. Remove resolve if pdf capability needs to be added
+				let response = {
+					categories: categories,
+					tasks: tasksReport,
+					projects: projectReport,
+				}
+				return resolve({
+					success: true,
+					message: CONSTANTS.apiResponses.REPORTS_GENERATED,
+					data: {
+						dataAvailable: true,
+						data: response,
+					},
+				})
+
+				if (getPdf == true) {
+					let reportTaskData = {}
+					Object.keys(tasksReport).map((taskData) => {
+						reportTaskData[UTILS.camelCaseToTitleCase(taskData)] = tasksReport[taskData]
+					})
+
+					let categoryData = {}
+					Object.keys(categories).map((category) => {
+						categoryData[UTILS.camelCaseToTitleCase(category)] = categories[category]
+					})
+
+					let types = await this.types()
+					let returnTypeInfo = types.data.filter((type) => {
+						if (type.value == reportType) {
+							return type.label
+						}
+					})
+
+					let pdfRequest = {
+						reportType: returnTypeInfo[0].label,
+						sharedBy: userName,
+						reportTitle: pdfReportString,
+						categories: categoryData,
+						tasks: reportTaskData,
+						projects: projectReport,
+					}
+					if (programId != '') {
+						pdfRequest['programName'] = projectDetails[0].programInformation.name
+					}
+					if (entityId != '') {
+						pdfRequest['entityName'] = projectDetails[0].entityInformation.name
+					}
+
+					//send data to report service to generate PDF.
+					let response = await reportService.entityReport(userToken, pdfRequest)
+					if (response && response.success == true) {
+						return resolve({
+							success: true,
+							message: CONSTANTS.apiResponses.REPORT_GENERATED,
+							data: {
+								data: {
+									downloadUrl: response.data.pdfUrl,
+								},
+							},
+						})
+					}
+				} else {
+					let response = {
+						categories: categories,
+						tasks: tasksReport,
+						projects: projectReport,
+					}
+					return resolve({
+						success: true,
+						message: CONSTANTS.apiResponses.REPORTS_GENERATED,
+						data: {
+							dataAvailable: true,
+							data: response,
+						},
+					})
+				}
+			} catch (error) {
+				return resolve({
+					success: false,
+					message: error.message,
+					data: false,
+				})
+			}
+		})
+	}
+
+	/**
+	 * Generates a PDF report based on entity report data.
+	 * @param {Object} entityReportData - Data for generating the report.
+	 * @param {string} userId - User ID associated with the report generation.
+	 * @returns {Promise<Object>} A promise that resolves to an object containing PDF generation status and URL.
+	 */
+	unnatiEntityReportPdfGeneration = async function (entityReportData) {
```

Comment:
@MallanagoudaB is this function similar to the one we have in common_handler. Where this function used?

Reply:
its the same function i removed it and no code break after this